authorizations/authorization-jdbc/src/main/java/org/apache/gravitino/authorization/jdbc/JdbcSecurableObjectMappingProvider.java

```diff
@@ -0,0 +1,213 @@
+  @Override
+  public Set<AuthorizationPrivilege> ownerMappingRule() {
+    return ownerPrivileges;
+  }
+
+  @Override
+  public Set<Privilege.Name> allowPrivilegesRule() {
+    return privilegeMapping.keySet();
+  }
+
+  @Override
+  public Set<MetadataObject.Type> allowMetadataObjectTypesRule() {
+    return allowObjectTypes;
+  }
+
+  @Override
+  public List<AuthorizationSecurableObject> translatePrivilege(SecurableObject securableObject) {
+    List<AuthorizationSecurableObject> authObjects = Lists.newArrayList();
+    List<AuthorizationPrivilege> databasePrivileges = Lists.newArrayList();
+    List<AuthorizationPrivilege> tablePrivileges = Lists.newArrayList();
+    JdbcAuthorizationObject databaseObject;
+    JdbcAuthorizationObject tableObject;
+    switch (securableObject.type()) {
+      case METALAKE:
+      case CATALOG:
+        convertJdbcPrivileges(securableObject, databasePrivileges, tablePrivileges);
+
+        if (!databasePrivileges.isEmpty()) {
+          databaseObject =
+              new JdbcAuthorizationObject(JdbcAuthorizationObject.ALL, null, databasePrivileges);
+          authObjects.add(databaseObject);
+        }
+
+        if (!tablePrivileges.isEmpty()) {
+          tableObject =
+              new JdbcAuthorizationObject(
+                  JdbcAuthorizationObject.ALL, JdbcAuthorizationObject.ALL, tablePrivileges);
+          authObjects.add(tableObject);
+        }
+        break;
+
+      case SCHEMA:
+        convertJdbcPrivileges(securableObject, databasePrivileges, tablePrivileges);
+        if (!databasePrivileges.isEmpty()) {
+          databaseObject =
+              new JdbcAuthorizationObject(securableObject.name(), null, databasePrivileges);
+          authObjects.add(databaseObject);
+        }
+
+        if (!tablePrivileges.isEmpty()) {
+          tableObject =
+              new JdbcAuthorizationObject(
+                  securableObject.name(), JdbcAuthorizationObject.ALL, tablePrivileges);
+          authObjects.add(tableObject);
+        }
+        break;
+
+      case TABLE:
+        convertJdbcPrivileges(securableObject, databasePrivileges, tablePrivileges);
+        if (!tablePrivileges.isEmpty()) {
+          MetadataObject metadataObject =
+              MetadataObjects.parse(securableObject.parent(), MetadataObject.Type.SCHEMA);
+          tableObject =
+              new JdbcAuthorizationObject(
+                  metadataObject.name(), securableObject.name(), tablePrivileges);
+          authObjects.add(tableObject);
+        }
+        break;
+
+      default:
+        throw new IllegalArgumentException(
+            String.format("Don't support metadata object type %s", securableObject.type()));
+    }
+
+    return authObjects;
+  }
+
+  @Override
+  public List<AuthorizationSecurableObject> translateOwner(MetadataObject metadataObject) {
+    List<AuthorizationSecurableObject> objects = Lists.newArrayList();
+    JdbcPrivilege allPri = JdbcPrivilege.valueOf(JdbcPrivilege.Type.ALL);
```

Comment:
I think it better to change `allPri` to `allPrivileges`?

Reply:
Renamed.